sgkit/io/vcf/vcf_writer.py

```diff
@@ -26,16 +29,73 @@
+    "AD": "Total read depth for each allele",
+    "ADF": "Read depth for each allele on the forward strand",
+    "ADR": "Read depth for each allele on the reverse strand",
+    "AF": "Allele frequency for each ALT allele in the same order as listed",
+    "AN": "Total number of alleles in called genotypes",
+    "BQ": "RMS base quality",
+    "CIGAR": "Cigar string describing how to align an alternate allele to the reference allele",
+    "DB": "dbSNP membership",
+    "DP": "Combined depth across samples",
+    "END": "End position on CHROM",
+    "H2": "HapMap2 membership",
+    "H3": "HapMap3 membership",
+    "MQ": "RMS mapping quality",
+    "MQ0": "Number of MAPQ == 0 reads",
+    "NS": "Number of samples with data",
+    "SB": "Strand bias",
+    "SOMATIC": "Somatic mutation",
+    "VALIDATED": "Validated by follow-up experiment",
+    "1000G": "1000 Genomes membership",
+}
+
+# [Table 2: Reserved genotype keys]
+RESERVED_FORMAT_KEY_DESCRIPTIONS = {
+    "AD": "Read depth for each allele",
+    "ADF": "Read depth for each allele on the forward strand",
+    "ADR": "Read depth for each allele on the reverse strand",
+    "DP": "Read depth",
+    "EC": "Expected alternate allele counts",
+    "FT": 'Filter indicating if this genotype was "called"',
+    "GL": "Genotype likelihoods",
+    "GP": "Genotype posterior probabilities",
+    "GQ": "Conditional genotype quality",
+    "GT": "Genotype",
+    "HQ": "Haplotype quality",
+    "MQ": "RMS mapping quality",
+    "PL": "Phred-scaled genotype likelihoods rounded to the closest integer",
+    "PP": "Phred-scaled genotype posterior probabilities rounded to the closest integer",
+    "PQ": "Phasing quality",
+    "PS": "Phase set",
+}
+
 
 def write_vcf(
-    input: Dataset,
-    output: Union[PathType, TextIO],
+    input: Dataset, output: Union[PathType, TextIO], *, vcf_header: Optional[str] = None
 ) -> None:
     """Convert a dataset to a VCF file.
 
-    The VCF fields included in the output are those in the ``vcf_header``
-    attribute of the dataset. There is currently no way to change the fields
-    that are included apart from by manually updating this attribute.
+    The VCF header to use is dictated by the ``vcf_header`` parameter, and the
+    ``vcf_header`` attribute on the input dataset. The ``vcf_header`` parameter
+    takes precedence and will be used if specified. Any variables in the dataset
+    that are not in the header will not be included in the output.
+
+    Otherwise, if the ``vcf_header`` attribute is present (because the dataset
+    derives from one imported via :func:`vcf_to_zarr`) it will be used to
+    generate the new VCF header. Any variables in the dataset that are not
+    specified in the original header will have corresponding header lines
+    added, and any variables not in the dataset that are in the original
+    header will have the corresponding header line omitted.
```

Comment:
I found this hard to parse - how about:

```suggestion
    The VCF header to use is dictated by either the ``vcf_header`` parameter or the
    ``vcf_header`` attribute on the input dataset. 

    If specified, the ``vcf_header`` parameter will be used, and any variables in the dataset
    that are not in this header will not be included in the output.

    If the ``vcf_header`` parameter is left as the default (`None`) and a ``vcf_header`` 
    attribute is present in the dataset (such as one created by :func:`vcf_to_zarr`), 
    it will be used to generate the new VCF header. In this case, any variables in the
    dataset that are not specified in this header will have corresponding header lines
    added, and any lines in the header without a corresponding variable in the dataset
    will be omitted.
```